Capabilities 2021

christianliebel opened the pull request:
Closes #2152 

- [x] Frontmatter
- [x] Feedback from @hemanth https://github.com/christianliebel/web-almanac-2021-capabilities/commit/8740fa6e5d4214a8fb3443a80ed869f233e7e767
- [x] Visualizations
- [x] Screenshots
- [x] Add discussion ID (blocked by #2414)
- [x] Resolve all remarks

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,228 @@
+Out of all 6,286,373 desktop and 7,491,840 mobile websites in the HTTP Archive, the File System Access API is used on 29 desktop and 23 mobile sites. Examples for those sites are the image editor [Excalidraw](https://excalidraw.com/), which allows you to sketch diagrams in a hand-drawn look and save them to the disk. Another example is [CorelDRAW.app](https://coreldraw.app/), a web version of the ...
+
+### Async Clipboard API
+The [Async Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) allows you to read and write data from or to the clipboard. Due to its asynchronous nature, it enables use cases like scaling down an image while pasting it—all without blocking the UI. It replaces less capable APIs like `document.execCommand()` that were previously used to interact with the clipboard.
```
I think it's the em-dash?!

christianliebel replied:
GitHub tells me the "suggestion is invalid because no changes were made to the code"…

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+```
+
+#### Read Access
+Similar to copying data to the clipboard, there are two methods to paste data back from the clipboard: First, another [shorthand method called `readText()`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText) that returns plain text from the clipboard. Using the [`read()` method](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read), you access all items in the clipboa ...
````
```suggestion
Similar to copying data to the clipboard, there are two methods to paste data back from the clipboard: First, another shorthand method called  [`readText()`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText) that returns plain text from the clipboard. Using the [`read()`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) method, you access all items in the clipboard in the data formats supported by the browser.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+With 560,359 (8.91%) desktop and 618,062 (8.25%) mobile sites, the Async Clipboard API (`writeText()` method) is one of the most used Fugu APIs. The `write()` method is used on 1,180 desktop and 1,227 mobile sites. The commercial website [Clipping Magic](https://clippingmagic.com/) allows you to remove the background of an image with the help of an AI algorithm. Just paste an image from the clipb ...
+
+### Web Share API
+The [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) allows you to share text, a URL, or files from a website or web application with other applications, e.g., mail clients or messengers. To do so, call the [`navigator.share()` method](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share). It takes an object with the data to share with another applicat ...
```
```suggestion
The [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) allows you to share text, a URL, or files from a website or web application with other applications, e.g., mail clients or messengers. To do so, call the [`navigator.share()`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) method. It takes an object with the data to share with another application. The browser then opens the built-in share sheet, where the user can select the target application from. The method returns a promise that resolves in case the content was successfully shared; otherwise, it will be rejected.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+})
+``` 
+
+The Web Share API is supported by Safari (iOS, macOS), and Chrome and Edge (Windows, Chrome OS). It’s currently a [Working Draft](https://www.w3.org/TR/web-share/) at the Web Applications Working Group. This is one of the first stages of the track to becoming a W3C Recommendation.
````
```suggestion
The Web Share API is supported by Safari (iOS, macOS), and Chrome and Edge (Windows, Chrome OS). It's currently a [Working Draft](https://www.w3.org/TR/web-share/) at the Web Applications Working Group. This is one of the first stages of the track to becoming a W3C Recommendation.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+}
+```
+
+If you want to register for origins other than your web app’s origin, you need to [verify your ownership of them](https://web.dev/pwa-url-handler/#the-web-app-origin-association-file). The capability is at a relatively early stage: It’s only supported on Chrome and Edge on the desktop. URL Handling is currently available as an Origin Trial. This means that the capability is not generally availabl ...
````
```suggestion
If you want to register for origins other than your web app's origin, you need to [verify your ownership of them](https://web.dev/pwa-url-handler/#the-web-app-origin-association-file). The capability is at a relatively early stage: It's only supported on Chrome and Edge on the desktop. URL Handling is currently available as an Origin Trial. This means that the capability is not generally available yet. Instead, developers need to register for an Origin Trial token first and deliver this token along with their website to use this capability. You can find more information in the [Origin Trials Guide for Web Developers](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+2. `new-client` opens a new window for the PWA
+3. `existing-client-navigate`  navigates an existing client to the new URL or opens a new window if no client exists
+
+Declarative Link Capturing also is an extension of the Web Application Manifest. To use it, you need to add the `capture_links` property to your manifest. This property takes a string or an array of strings matching the three modes from above. If you use an array, the browser will fall back to the next entry if it doesn’t support a particular mode.
```
```suggestion
Declarative Link Capturing also is an extension of the Web Application Manifest. To use it, you need to add the `capture_links` property to your manifest. This property takes a string or an array of strings matching the three modes from above. If you use an array, the browser will fall back to the next entry if it doesn't support a particular mode.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+}
+```
+
+This capability is at an early stage as well. It is only supported on Chrome OS. Currently, 36 desktop sites and 11 mobile sites use this capability, for example, [Periodex](https://periodex.co/), a PWA showing the periodic table of elements. This app uses the `capture_links` configuration as shown in the listing above: If supported, the browser should reuse the existing window, otherwise, open a ...
````
```suggestion
This capability is at an early stage as well. It is only supported on Chrome OS. Currently, 36 desktop sites and 11 mobile sites use this capability, for example, [Periodex](https://periodex.co/), a PWA showing the periodic table of elements. This app uses the `capture_links` configuration as shown in the listing above: If supported, the browser should reuse the existing window, otherwise, open a new one, and if that's not supported, it should behave as normal.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+The next set of capabilities focuses on hardware-related APIs. In Chromium-based browsers, there are many APIs to access hardware interfaces, including but not limited to USB, Bluetooth, and serial devices. Furthermore, the Generic Sensor API allows you to read from device sensors. All capabilities discussed in this section are only available on Chromium-based browsers and on systems where the re ...
+
+#### Web USB API
+The [Web USB API](https://web.dev/usb/) allows developers to access USB devices without any drivers or third-party applications. For instance, this capability is interesting for firmware updaters that developers otherwise would have to implement as separate native apps for different platforms. You need to call the `navigator.usb.requestDevice()` method to access USB devices. It takes an object wh ...
```
```suggestion
The [Web USB API](https://web.dev/usb/) allows developers to access USB devices without any drivers or third-party applications. For instance, this capability is interesting for firmware updates that developers otherwise would have to implement as separate platform-specific apps for different platforms. You need to call the `navigator.usb.requestDevice()` method to access USB devices. It takes an object which defines filters for the list of all connected USB devices. You need to specify the `vendorId` at least. The browser shows a device picker where the user can choose a matching device. From there, you can begin a device session.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+}
+```
+
+The API is generally available on Chromium-based browsers since version 61. 182 desktop and 155 mobile sites use this API, for example, the PWA [Vysor](https://app.vysor.io/#/) that allows you to mirror the screen of an Android or iOS device—all without installing any additional software on your computer.
````
```suggestion
The API is generally available on Chromium-based browsers since version 61. 182 desktop and 155 mobile sites use this API, for example, the PWA [Vysor](https://app.vysor.io/#/) that allows you to mirror the screen of an Android or iOS device—all without installing any additional software on your computer.
```

christianliebel replied:
GitHub tells me the "suggestion is invalid because no changes were made to the code"…

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+The API is generally available on Chromium-based browsers on Chrome OS, Android, macOS, and Windows starting from version 56. On Linux, the API is provided behind a flag. 71 desktop and 45 mobile sites make use of this capability. For instance, the [Brewfather](https://web.brewfather.app/) PWA targeted at home-brewers allows them to send a beer recipe wirelessly over to a Bluetooth-enabled brewin ...
+
+#### Web Serial API
+The [Web Serial API](https://web.dev/serial/) allows you to connect with serial devices such as microcontrollers. To do so, call the `navigator.serial.requestPort()` method. You can optionally pass in a method to filter the device list. The browser shows a device picker where the user can choose a device. Next, you can open the connection by calling the port’s `open()` method.
```
```suggestion
The [Web Serial API](https://web.dev/serial/) allows you to connect with serial devices such as microcontrollers. To do so, call the `navigator.serial.requestPort()` method. You can optionally pass in a method to filter the device list. The browser shows a device picker where the user can choose a device. Next, you can open the connection by calling the port's `open()` method.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+}
+```
+
+This capability is relatively new, as it shipped with Chromium 89 in March 2021. Currently, 15 desktop and 14 mobile sites use Web Serial API, including the [Duino App](https://duino.app/) that allows you to develop programs for Arduino and ESP microcontrollers right in your browser. They are compiled on a remote server and then uploaded to a connected board via the Web Serial API.
````
```suggestion
This capability is relatively new, as it shipped with Chromium 89 in March 2021. Currently, 15 desktop and 14 mobile sites use the Web Serial API, including the [Duino App](https://duino.app/) that allows you to develop programs for Arduino and ESP microcontrollers right in your browser. They are compiled on a remote server and then uploaded to a connected board via the Web Serial API.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+This capability is relatively new, as it shipped with Chromium 89 in March 2021. Currently, 15 desktop and 14 mobile sites use Web Serial API, including the [Duino App](https://duino.app/) that allows you to develop programs for Arduino and ESP microcontrollers right in your browser. They are compiled on a remote server and then uploaded to a connected board via the Web Serial API.
+
+#### Generic Sensor API
+Finally, the [Generic Sensor API](https://web.dev/generic-sensor/) allows you to read sensor data from the device’s sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onreading` and `onerror` even ...
```
```suggestion
Finally, the [Generic Sensor API](https://web.dev/generic-sensor/) allows you to read sensor data from the device's sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onreading` and `onerror` events, you can get notified for updated sensor values, or errors respectively. Finally, you need to start the reading by calling the `start()` method.
```

tomayac commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,234 @@
+}
+```
+
+The capability is supported by Chromium browsers starting from version 67. The relative orientation sensor is used by 824 desktop and 831 mobile sites, the linear acceleration sensor by 257 desktop and 237 mobile sites, and the gyroscope by 36 desktop and 22 mobile sites. An example application that uses all three of them is [VDO.Ninja](https://obs.ninja/), the former OBS Ninja. This software all ...
````
```suggestion
The capability is supported by Chromium browsers starting from version 67. The relative orientation sensor is used by 824 desktop and 831 mobile sites, the linear acceleration sensor by 257 desktop and 237 mobile sites, and the gyroscope by 36 desktop and 22 mobile sites. An example application that uses all three of them is [VDO.Ninja](https://obs.ninja/), the former OBS Ninja. This software allows you to remotely connect with video broadcasting software such as OBS. The app allows the connected broadcasting software to read sensor data from the device. For example, to capture a smartphone's movements when streaming virtual reality content.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+
+The capability is supported by Chromium browsers starting from version 67. The relative orientation sensor is used by 824 desktop and 831 mobile sites, the linear acceleration sensor by 257 desktop and 237 mobile sites, and the gyroscope by 36 desktop and 22 mobile sites. An example application that uses all three of them is [VDO.Ninja](https://obs.ninja/), the former OBS Ninja. This software all ...
+
+### Sites using most capabilities
```
```suggestion
### Sites using the most capabilities
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+The capability is supported by Chromium browsers starting from version 67. The relative orientation sensor is used by 824 desktop and 831 mobile sites, the linear acceleration sensor by 257 desktop and 237 mobile sites, and the gyroscope by 36 desktop and 22 mobile sites. An example application that uses all three of them is [VDO.Ninja](https://obs.ninja/), the former OBS Ninja. This software all ...
+
+### Sites using most capabilities
+The analysis also identified the websites using the most capabilities from the HTTP Archive data set. The crawler is capable of identifying 30 Fugu APIs in total.
```
```suggestion
The analysis also identified the websites using the most capabilities from the HTTP Archive data set. The detection script is capable of identifying 30 Fugu APIs in total.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+### Sites using most capabilities
+The analysis also identified the websites using the most capabilities from the HTTP Archive data set. The crawler is capable of identifying 30 Fugu APIs in total.
+
+1. The number one site is [whatwebcando.today](https://whatwebcando.today/), which uses 28 capabilities. It showcases different HTML5 device integration APIs by providing a live demo for every capability. Naturally, the number of used APIs is very high. In the result set, a similar site called [whatpwacando.today](https://whatpwacando.today/)  showcases PWA capabilities and uses eight APIs.
```
```suggestion
1. The number one site is [whatwebcando.today](https://whatwebcando.today/), which uses 28 capabilities. It showcases different HTML5 device integration APIs by providing a live demo for every capability. Naturally, the number of used APIs is very high. In the result set, a similar site called [whatpwacando.today](https://whatpwacando.today/) showcases PWA capabilities and uses eight APIs.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+The analysis also identified the websites using the most capabilities from the HTTP Archive data set. The crawler is capable of identifying 30 Fugu APIs in total.
+
+1. The number one site is [whatwebcando.today](https://whatwebcando.today/), which uses 28 capabilities. It showcases different HTML5 device integration APIs by providing a live demo for every capability. Naturally, the number of used APIs is very high. In the result set, a similar site called [whatpwacando.today](https://whatpwacando.today/)  showcases PWA capabilities and uses eight APIs.
+2. The [PolisNotis](https://polisnotis.se/) PWA shows police notices in Sweden. It uses ten APIs, including the Declarative Link Capturing API to define that the PWA should always open a new window when clicking a PWA-related link. The Web Share API is used in the source code, but the sharing functionality is not exposed to the UI.
+3. The website [System Scanner](https://system-scanner.net/) uses nine APIs: It shows an overview of the system information exposed by the browser, including sensor information provided by the Generic Sensor API.
+4. Eight sites use eight Fugu APIs: One of them is the aforementioned [Excalidraw](https://excalidraw.com/), an online drawing tool for creating drawings in a hand-drawn style. As a traditional productivity app, it benefits from the new capabilities.
```
```suggestion
1. The [PolisNotis](https://polisnotis.se/) PWA shows police notices in Sweden. It uses ten APIs, including the Declarative Link Capturing API to define that the PWA should always open a new window when clicking a PWA-related link. The Web Share API is used in the source code, but the sharing functionality is not exposed to the UI. The app also uses the Badging API to alert the user via the app icon if there is a new notice.
1. The website [System Scanner](https://system-scanner.net/) uses nine APIs: It shows an overview of the system information exposed by the browser, including sensor information provided by the Generic Sensor API.
1. Eight sites use eight Fugu APIs: One of them is the aforementioned [Excalidraw](https://excalidraw.com/), an online drawing tool for creating drawings in a hand-drawn style. As a traditional productivity app, it benefits from the new capabilities.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+3. The website [System Scanner](https://system-scanner.net/) uses nine APIs: It shows an overview of the system information exposed by the browser, including sensor information provided by the Generic Sensor API.
+4. Eight sites use eight Fugu APIs: One of them is the aforementioned [Excalidraw](https://excalidraw.com/), an online drawing tool for creating drawings in a hand-drawn style. As a traditional productivity app, it benefits from the new capabilities.
+
+The results also include sites that aren’t proactively using the APIs. For example, some sites ship library code that could theoretically access the capabilities. Some sites check for the presence of Fugu APIs to determine the user’s browser.
```
```suggestion
The results also include sites that aren't proactively using the APIs. For example, some sites ship library code that could theoretically access the capabilities. Some sites check for the presence of Fugu APIs to determine the user’s browser.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+The results also include sites that aren’t proactively using the APIs. For example, some sites ship library code that could theoretically access the capabilities. Some sites check for the presence of Fugu APIs to determine the user’s browser.
+
+### Conclusion
+Capabilities help move the web forward by unlocking more and more use cases for developers. As this chapter shows, developers use the new web platform APIs to build powerful applications. In contrast to their native counterparts, those applications don’t need to be installed to the system and don’t require any additional third-party runtimes or plug-ins to work. They run on any platform that can  ...
```
```suggestion
Capabilities help move the web forward by unlocking more and more use cases for developers. As this chapter shows, developers use the new web platform APIs to build powerful applications. In contrast to their platform-specific counterparts, those applications don't necessarily need to be installed to the system and don't require any additional third-party runtimes or plug-ins to work. They run on any platform that can run a powerful browser.
```

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+### Conclusion
+Capabilities help move the web forward by unlocking more and more use cases for developers. As this chapter shows, developers use the new web platform APIs to build powerful applications. In contrast to their native counterparts, those applications don’t need to be installed to the system and don’t require any additional third-party runtimes or plug-ins to work. They run on any platform that can  ...
+
+One example of this concept working is the Visual Studio Code IDE. This application has always been web-based, but it still relied on native application wrappers like Electron. Thanks to capabilities like the File System Access API, Microsoft was able to release the application as a browser application ([vscode.dev](https://vscode.dev)) in October 2021. Almost all IDE features work here, except a ...
```
```suggestion
One example of this concept working is the Visual Studio Code IDE. This application has always been web-based, but it still relied on platform-specific application wrappers like Electron. Thanks to capabilities like the File System Access API, Microsoft was able to release the application as a browser application ([vscode.dev](https://vscode.dev)) in October 2021. Almost all IDE features work here, except access to the terminal, since there is no capability for this yet.
```

tomayac reviewed (commented):
I like where this is going.

Do you think we can get a number of screenshots for the examples you mention in each section? This would help feel the article a lot more "interactive".

Can we also get the forum software Discourse ([example](https://discourse.mozilla.org/t/getters-setters-and-documenting-on-mdn/29338)) into the list at the end? It uses 8 very interesting Fugu APIs and powers a bunch of forums on the web.

tomayac commented:
OK, finished the review of the current iteration. One thing to mention is that "native" belongs to the to-be-avoided words, so I try to reformulate where necessary. Also replaced curly apostrophes and quotes.

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
-authors: []
-reviewers: []
-analysts: []
+description: Capabilities give web apps access to hardware interfaces, enhance web-based productivity apps, and more.
```
Can we start this in the standard way:

```suggestion
description: description: Capabilities chapter of the 2020 Web Almanac covering brand-new, powerful web platform APIs that give web apps access to hardware interfaces, enhance web-based productivity apps, and more.
```

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
-reviewers: []
-analysts: []
+description: Capabilities give web apps access to hardware interfaces, enhance web-based productivity apps, and more.
+authors: [christianliebel]
+reviewers: [tomayac, hemanth]
+analysts: [tomayac]
```
Reminder to copy these people's details to `src/config/2021.json` as part of this PR. All were involved last year so should just be a copy and paste from that and an update to the roles.

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+analysts: [tomayac]
 editors: []
 translators: []
 results: https://docs.google.com/spreadsheets/d/1b4moteB9EiLYkH1Ln9qfi1tnU-E4N2UQ87uayWytDKw/
```
Need a discussion id here (not assigned yet - see ): #2414 

```suggestion
discuss: ????
results: https://docs.google.com/spreadsheets/d/1b4moteB9EiLYkH1Ln9qfi1tnU-E4N2UQ87uayWytDKw/
```

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
 ---
 
-## TODO
+## Capabilities
```
The H1 heading will automatically be added so should start this with:

```md
## Introduction
```
and end it with:

```md
## Conclusion
```

So I think all your headings might be 1 level too deep?

christianliebel commented:
@tomayac Cool, thanks. I accepted all suggestions. I'll go and add the visualizations next.

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,234 @@
+## Capabilities
+Capabilities are new web platform APIs that unlock entirely new use cases for web applications. Those new APIs are essential for [Progressive Web Apps (PWA)](../pwa), a web-based application model. A PWA is a web app that users can install to their system. PWAs run even offline and launch quickly. To integrate with the underlying operating system, PWAs can only use web platform APIs. While browse ...
+
+The [Capabilities Project](https://www.chromium.org/teams/web-capabilities-fugu) (codename Fugu) is a joint effort by Microsoft, Intel, Google, and other Chromium contributors. It tries to bridge the gap between platform-specific applications and web apps by designing and implementing new powerful web platform APIs in a secure and privacy-preserving manner. As capabilities unlock more and more us ...
```
We've actually switched to using HTML links for external links so we can add a `hreflang="en"` so the translations get a little `(en)` symbol when a translation doesn't exist:

```suggestion
The <a hreflang="en" href="https://www.chromium.org/teams/web-capabilities-fugu">Capabilities Project</a> (codename Fugu) is a joint effort by Microsoft, Intel, Google, and other Chromium contributors. It tries to bridge the gap between platform-specific applications and web apps by designing and implementing new powerful web platform APIs in a secure and privacy-preserving manner. As capabilities unlock more and more use cases, they lay the path for entire new application categories to finally make the shift to the web (e.g., IDEs, image editors, or office applications). Over the last two years, the focus for the Fugu team has been on capabilities for desktop productivity applications and hardware-related APIs. This chapter briefly introduces several new capabilities and analyzes how many different desktop and mobile websites use them. As capabilities are particularly interesting for app-like websites, their relative usage is comparatively low. This is why absolute website numbers are used in this chapter. For each capability, there will be a demo website or app that makes use of it.
```

Note wikipedia and MDN links don't need this and can use markdown as we pick up the fact they have `en` in their URLs, so can continue to use markdown link syntax for them.

https://github.com/HTTPArchive/almanac.httparchive.org/blob/78ea8ea14f748a9f88d22cfcb48700eaf4bc8ced/src/static/css/almanac.css#L1559-L1587

christianliebel replied:
@tunetheweb How to deal with the `[Progressive Web Apps (PWA)](../pwa)` link? Does that need to be an HTML link too? I converted the other links (except the MDN ones).

tunetheweb replied:
We leave internal links as they are. In the overly optimistic hope that they will be translated to every language 😁

Technically the link (since it is relative) is to the translated version, but in reality it will redirected to English most of the time, since a translated version will often not exist. 😞

christianliebel replied:
Cool, thanks!

tunetheweb reviewed (changes requested):
Don't mean to interrupt your writing flow, but spotted a few small technical things to fix at some point before taking this out of Draft.

christianliebel commented:
@tomayac I added a paragraph on Discourse, some screenshots, and big numbers as visualizations. Google Sheets does _not_ support log scales with values < 1, so that's not an option, unfortunately. Furthermore, I added Photoshop as another example for apps using capabilities to the conclusion.

@tunetheweb Can we have a test deployment of this chapter?

tunetheweb commented:
> @tunetheweb Can we have a test deployment of this chapter?

Here you go: https://20211101t183423-dot-webalmanac.uk.r.appspot.com

christianliebel commented:
Looks great, thanks!

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+
+Some websites from the result set are Internet forums based on <a hreflang="en" href="https://www.discourse.org/">Discourse</a>. This forum software supports a total of eight Fugu APIs. Discourse-based forums are installable and support, among others, the Badging API to show the amount of unread notifications.
+
+The results also include sites that aren't proactively using the APIs. For example, some sites ship library code that could theoretically access the capabilities. Some sites check for the presence of Fugu APIs to determine the user's browser.
```
I don't understand that last sentence? Do you mean they are using these APIs to fingerprint the user? Is that a risk of these APIs? Should we discuss that? Should we link to the Privacy chapter?

Also how does this work with the aforementioned need of interaction to use these APIs? Or is it allowed to test if the API exists (`if genericAPI in window`) without needing that interaction?

On a similar note to any of these need permission prompts before using? I would hope so! Should that be mentioned somewhere?

christianliebel replied:
> I don't understand that last sentence? Do you mean they are using these APIs to fingerprint the user? Is that a risk of these APIs? Should we discuss that? Should we link to the Privacy chapter?

Some of the sites I've seen test for the presence of certain APIs to determine the browser brand without looking at the user agent. I don't think this is considered fingerprinting. All specification texts contain a section about security and privacy considerations, we could mention that in the introduction. And yes, we can link to the privacy chapter, looks like they have a section on permissions.

> Also how does this work with the aforementioned need of interaction to use these APIs? Or is it allowed to test if the API exists (if genericAPI in window) without needing that interaction?

All APIs presented in this chapter can be detected without user interaction.

> On a similar note do any of these need permission prompts before using? I would hope so! Should that be mentioned somewhere?

Most APIs have implicit permission prompts (e.g., file picker, device picker, …), and it's always noted in the section about the API. Access to motion and light sensor APIs is granted by default.

tunetheweb replied:
OK let's leave as is. Re-reading it it's fine. Resolving.

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+## Conclusion
+Capabilities help move the web forward by unlocking more and more use cases for developers. As this chapter shows, developers use the new web platform APIs to build powerful applications. In contrast to their platform-specific counterparts, those applications don't necessarily need to be installed to the system and don't require any additional third-party runtimes or plug-ins to work. They run on ...
+
+One example of this concept working is Visual Studio Code. This application has always been web-based, but it still relied on platform-specific application wrappers like Electron. Thanks to capabilities like the File System Access API, Microsoft was able to release the application as a browser application (<a hreflang="en" href="https://vscode.dev">vscode.dev</a>) in October 2021. Almost all feat ...
```
Maybe hint at the future?

```suggestion
One example of this concept working is Visual Studio Code. This application has always been web-based, but it still relied on platform-specific application wrappers like Electron. Thanks to capabilities like the File System Access API, Microsoft was able to release the application as a browser application (<a hreflang="en" href="https://vscode.dev">vscode.dev</a>) in October 2021. Almost all features work here, except debugging or terminal access, since there is no capability for this (yet!).
```

christianliebel replied:
I don't expect these APIs to land in the near future, but yes, we can go with that.

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+
+One example of this concept working is Visual Studio Code. This application has always been web-based, but it still relied on platform-specific application wrappers like Electron. Thanks to capabilities like the File System Access API, Microsoft was able to release the application as a browser application (<a hreflang="en" href="https://vscode.dev">vscode.dev</a>) in October 2021. Almost all feat ...
+
+Another example is <a hreflang="en" href="https://photoshop.adobe.com">Adobe Photoshop</a>, which was also released as a web application in October 2021. Photoshop also uses several of the capabilities presented here, as well as WebAssembly, to migrate existing code to the web. Thus, the Capabilities project paves the way for entire categories of applications to finally migrate to the web.
```
Photoshop got most of the hype, but it's also Illustrator.

Also worth linking to this article?: https://web.dev/ps-on-the-web/

Also worth mentioning that Adobe don't plan to stop there?: https://web.dev/ps-on-the-web/#what's-next-for-adobe-on-the-web

tunetheweb reviewed (changes requested):
Fascinating read!

Left some feedback and suggestions.

One thing worth considering is that it does read a little same: API description, code, usage. So it comes across as a bit more of a tutorial than the Web Almanac normally is. But that may just be the nature of the topic? The fact that all the stats are just big figures adds to that impression. I've suggested one place where you might be able to use a chart for a change. Also last year you used a lot of the ChromeStatus charts - maybe should include one or two of them for any APIs with notable growth, to further add a bit of variety (along with the big figures)?

tomayac commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+Please note that most of the APIs presented here are so-called incubations. Unless noted, they are not (yet) W3C Recommendations, i.e., official web standards. Instead, these APIs are being worked on in the Web Platform Incubator Community Group (WICG), where browser vendors and developers can discuss new features. Some APIs have already shipped in several browsers; others are only available on C ...
+
+## File System Access API
+The first productivity-related API is the [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API). Web apps could already open files from the file system via `input[type=file]`: In a file picker, the user can choose one or more files to give the web application access to. Also, they could already save files to the Downloads folder via `a[download]`. The F ...
```
We could link to https://web.dev/browser-fs-access/#the-traditional-way-of-dealing-with-files where I have explained this in great detail.

tunetheweb replied:
Interesting that you went with the alternative notation in that @tomayac as suggested!

christianliebel replied:
Done!

christianliebel commented on src/config/2021.json:
```diff
@@ -245,6 +245,36 @@
       "avatar_url": "https://avatars1.githubusercontent.com/u/44442451?v=4&s=200",
       "github": "AbbyTsai"
     },
+    "christianliebel": {
+      "name": "Christian Liebel",
+      "teams": [
+        "authors"
+      ],
+      "avatar_url": "https://avatars1.githubusercontent.com/u/6698344?v=4&s=200",
+      "website": "https://christianliebel.com",
+      "github": "christianliebel",
+      "twitter": "christianliebel"
+    },
+    "hemanth": {
+      "name": "Hemanth HM",
+      "teams": [
+        "reviewers"
+      ],
+      "avatar_url": "https://avatars1.githubusercontent.com/u/18315?v=4&s=200",
+      "website": "https://h3manth.com",
+      "github": "hemanth",
+      "twitter": "gnumanth"
+    },
+    "tomayac": {
+      "name": "Thomas Steiner",
+      "teams": [
+        "analysts",
+        "reviewers"
+      ],
+      "avatar_url": "https://avatars1.githubusercontent.com/u/145676?v=4&s=200",
+      "website": "https://blog.tomayac.com/",
+      "github": "tomayac"
+    },
```
They should be ordered by the first name.

tunetheweb replied:
Yes but not in context of other names already in the file. E.g. "Barry Pollard" appears after "Thomas Steiner"

christianliebel replied:
Done!

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+
+{{ figure_markup(
+caption="Mobile websites use the File System Access API",
+content="7,491,840",
```
Yes. :D

christianliebel commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,368 @@
+
+Declarative Link Capturing also is an extension of the Web Application Manifest. To use it, you need to add the `capture_links` property to your manifest. This property takes a string or an array of strings matching the three modes from above. If you use an array, the browser will fall back to the next entry if it doesn't support a particular mode.
+
+```json
````
Done!

christianliebel commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,368 @@
+
+```js
+try {
+  const device = await navigator.usb.requestDevice({ filters: [{ vendorId: 0x8086 }] });
````
Done!

christianliebel commented:
@tunetheweb Thanks for your review comments, I already resolved some and will continue tomorrow.

tunetheweb commented:
> @tunetheweb Thanks for your review comments, I already resolved some and will continue tomorrow.

No bother. Hope you're finding them helpful but feel free to argue any back you don't agree with. Good news is that this is the editing stage complete too so once done here, you're done!

A few more changes I spotted:
- Can you add me (tunetheweb) as the editor? And also add editor to my list of roles in the 2021.json?
- Can you check headings are in sentence case? Most are fine as APIs (so should be Title Case for those words) but things like "Write Access" probable should be "Write access"
- Can you remove `"todo": "true"` from your chapter in 2021.json?

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+) }}
+
+### Web Bluetooth API
+The <a hreflang="en" href="https://web.dev/bluetooth/">Web Bluetooth API</a> allows you to communicate with nearby Bluetooth Low Energy devices using the Generic Attribute Profile (GATT). To find a matching device, call the `navigator.bluetooth.requestDevice()` method. In the following example, the list of Bluetooth devices is filtered by whether they offer a battery service or not. The browser s ...
```
BLE needs to be in caps: https://en.wikipedia.org/wiki/Bluetooth_Low_Energy

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+2. `new-client` opens a new window for the PWA
+3. `existing-client-navigate`  navigates an existing client to the new URL or opens a new window if no client exists
+
+Declarative Link Capturing also is an extension of the Web Application Manifest. To use it, you need to add the `capture_links` property to your manifest. This property takes a string or an array of strings matching the three modes from above. If you use an array, the browser will fall back to the next entry if it doesn't support a particular mode.
```
It's explained in the URL handling section directly above. I added an additional link there.

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+If you want to register for origins other than your web app's origin, you need to <a hreflang="en" href="https://web.dev/pwa-url-handler/#the-web-app-origin-association-file">verify your ownership of them</a>. The capability is at a relatively early stage: It's only supported on Chrome and Edge on the desktop. URL Handling is currently available as an Origin Trial. This means that the capability  ...
+
+{{ figure_markup(
+caption="Desktop websites use URL Handling",
+content="44",
```
Yes, desktop makes more sense for those APIs. The File System Access API, for example, is currently only supported on desktop. I've changed all big numbers to show the usage on desktop sites.

christianliebel commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,368 @@
+}
+```
+
+If you want to register for origins other than your web app's origin, you need to <a hreflang="en" href="https://web.dev/pwa-url-handler/#the-web-app-origin-association-file">verify your ownership of them</a>. The capability is at a relatively early stage: It's only supported on Chrome and Edge on the desktop. URL Handling is currently available as an Origin Trial. This means that the capability  ...
````
Rephrased the sentence as suggested, added the link, and left the last sentence in place. Please double check.

christianliebel commented on src/content/en/2021/capabilities.md:
````diff
@@ -1,20 +1,368 @@
+await writable.close();
+```
+
+### Read Access
````
Chrome Platform Status data shows that read is more used (= page loads) than write ([Read](https://chromestatus.com/metrics/feature/timeline/popularity/2369)/[Write](https://chromestatus.com/metrics/feature/timeline/popularity/2370)). In last year's chapter, writing was used more. I'd like to stick with addressing the writing first. I find it easier to follow how I can read something I've written before.

tunetheweb replied:
Absolutely fine. Resolving.

tunetheweb commented:
FYI ignore the "Test Website" CI error for now. I've a few extra steps to complete to allow that to work, which depends on which is the first chapter to go live. Let me know when all done and will add those if this is the first chapter to go live.

tunetheweb commented:
@christianliebel I've merged the necessary change into `main` to allow the CI to work on 2021 changes (#2428 ).

So once you merge `main` into this branch that should start working.

Note there is no need to rebase or squash commits, as we squash on merge to `main` anyway.

christianliebel commented:
@tunetheweb Thanks, I just rebased the changes from `main`. I'll continue tomorrow with resolving the other remarks, I guess only 6 of them are left.

tunetheweb commented:
Woo hoo!

![All checks have passed](https://user-images.githubusercontent.com/10931297/140302947-d1bc8f5e-0b57-496b-aef6-7cf51f3e335b.png)

rviscomi commented:
@christianliebel is the chapter peer reviewed and ready to merge now?

christianliebel commented:
@rviscomi Not yet, still need to resolve some comments, hope to do that later today. @tomayac should also give his +1 before we launch this.

tomayac commented:
Chris, could you create a preview with images, so it's easier to imagine the final result, please?

tunetheweb commented:
> Chris, could you create a preview with images, so it's easier to imagine the final result, please?

Chris can't create those - but I can!

Here's the latest: https://20211105t093205-dot-webalmanac.uk.r.appspot.com/en/2021/capabilities but be aware that there's a few unresolved comments above. Feel free to weigh in on those if you want.

tomayac commented:
Oh, awesome! Wow, this looks really great! Just small pieces of feedback:

- Could we break up the [wall of text](https://20211105t093205-dot-webalmanac.uk.r.appspot.com/en/2021/capabilities#:~:text=Capabilities%20are%20new,the%20browser%20settings.) in the intro a bit with subheadings and maybe sprinkle in the [Fugu fish](https://github.com/googlefonts/noto-emoji/blob/43f47be9404018cd9d8f73a227363a8f20acdab5/svg/emoji_u1f421.svg)?
- Can we flip the order and start with Clipboard and then File System Access so we have a high number (560,359) at the beginning and not a low number (29)?
- Do we have an image for the Generic Sensor APIs? A cool app using them?  
- Could we "dramatize" the [last section](https://20211105t093205-dot-webalmanac.uk.r.appspot.com/en/2021/capabilities#sites-using-the-most-capabilities)? Something like an "award for the most Fugu website" with a stupid trophy clipart 🏆 or a winner's podium where we montage screenshots of these websites on the stairs or something with medals 🥇🥈🥉? It's kind of fun to fake-compete on the most number of APIs used, but currently it comes across a little too dry for my taste. What do you think?

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+) }}
+
+### Generic Sensor API
+Finally, the <a hreflang="en" href="https://web.dev/generic-sensor/">Generic Sensor API</a> allows you to read sensor data from the device's sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onre ...
```
It's about the device's sensors (i.e., accelerometer, or gyroscope), I hoped the first sentence would make that clear. Would it help to rephrase it like this maybe?

```suggestion
While the previous APIs provide access to data transmission interfaces, the <a hreflang="en" href="https://web.dev/generic-sensor/">Generic Sensor API</a> allows you to read sensor data from the device's sensors, such as the accelerometer, gyroscope, or orientation sensor. To access a sensor, you create a new instance of a sensor class, e.g., `Accelerometer`. The constructor takes a configuration object with the requested frequency. By attaching to the `onreading` and `onerror` events, you can get notified for updated sensor values, or errors respectively. Finally, you need to start the reading by calling the `start()` method.
```

tunetheweb replied:
Ah I got you. So it's like devices and sensors, rather than direct serial port access to get to those devices.

Hmmmm.... can't think of a better way of phrasing it to be honest. Think I prefer your original wording. Will just leave it as is and resolve this.

christianliebel commented:
> One thing worth considering is that it does read a little same: API description, code, usage. So it comes across as a bit more of a tutorial than the Web Almanac normally is. But that may just be the nature of the topic?

@tunetheweb Yes, I guess that's the nature of the topic. Some of the APIs are brand new, so we can't assume the reader is familiar with them. Also, I want to be consistent in the description of the APIs.

> The fact that all the stats are just big figures adds to that impression. I've suggested one place where you might be able to use a chart for a change. Also last year you used a lot of the ChromeStatus charts - maybe should include one or two of them for any APIs with notable growth, to further add a bit of variety (along with the big figures)?

@tunetheweb Makes sense, I'll add one for the two "bigger" APIs Async Clipboard and Web Share.

> Could we break up the wall of text in the intro a bit with subheadings and maybe sprinkle in the Fugu fish?

@tomayac Done.

> Can we flip the order and start with Clipboard and then File System Access so we have a high number (560,359) at the beginning and not a low number (29)?

@tomayac Done.

christianliebel commented:
> Do we have an image for the Generic Sensor APIs? A cool app using them?

@tomayac We can use one of the Intel demos.

--- 

@tunetheweb The website lighthouse test is currently broken due to low contrast for "window" in code blocks, can you have a look?

tunetheweb commented:
> @tunetheweb The website lighthouse test is currently broken due to low contrast for "window" in code blocks, can you have a look?

Could you change include a change to out CSS in this PR?:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/1ccc53639cf9e33716c72c6875cef04b9ee3bc3a/src/static/css/almanac.css#L1495-L1497

From `#0086b3` to `#006080`

tunetheweb commented:
Sorry @christianliebel just accidental pushed an unrelated changed to this branch. Have reverted it now.

tunetheweb commented:
@christianliebel are you still working on this? Or is it all done?

I did have two unresolved comments about the really high usage of some of the APIs but, other than that, it looks like all comments are resolved so not sure if you were done?

I've, merged in `main` and re-staged the latest version here: https://20211110t003629-dot-webalmanac.uk.r.appspot.com/en/2021/capabilities

christianliebel commented:
> I did have two unresolved comments about the really high usage of some of the APIs but, other than that, it looks like all comments are resolved so not sure if you were done?

@tunetheweb I have these two comments and four other items on my list. I'm on vacation right now but will get to it within the next few days.

tunetheweb commented:
No probs. We’re in a great space with this chapter so enjoy your holiday and can pick this up when you’re back.

tunetheweb commented:
Not loving the Fugu fish to be honest, feels like it's just there fore the sake of it and a bit cartoony:

![image](https://user-images.githubusercontent.com/10931297/141646086-83b21948-63fe-44e3-a187-d01893f9fb5f.png)

But do recognise the desire to break up the text. The Structured Data chapter used a quote in their chapters which looks quite nice so that's another option if you have one of those:

![image](https://user-images.githubusercontent.com/10931297/141646146-cf1ece59-0044-4120-be46-45ec0e24ed9a.png)

Here's the markdown for that if you do want to use that:

```html
<figure>
<blockquote>_The Semantic Web_ is the name of a long-term project started by W3C with the stated purpose of realizing the idea of having data on the Web defined and linked in a way that it can be used by machines not just for display purposes, but for automation, integration, and reuse of data across various applications</blockquote>
<figcaption>— Greg Ross, <cite><a hreflang="en" href="https://www.techrepublic.com/article/an-introduction-to-tim-berners-lees-semantic-web/">An introduction to Tim Berners-Lee's Semantic Web</a></cite></figcaption>
</figure>
```

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+)
+}}
+
+With 566,049 (9.00%) desktop and 642,507 (8.58%) mobile sites, the Web Share API is the most used Fugu API. For example, the <a hreflang="en" href="https://beta.paintz.app/">beta version of the PaintZ app</a> allows you to share a drawing with another locally installed application via the save dialog.
```
I checked the first 10 sites from the sample URLs, and it appears that the high number can be traced back to a YouTube script (https://www.youtube.com/s/player/8d287e4d/player_ias.vflset/en_US/base.js) that is included when embedding a YouTube video. When the user clicks the Share button in the player, the Web Share API is called (if present).

.cc @tomayac

christianliebel commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,368 @@
+)
+}}
+
+With 560,359 (8.91%) desktop and 618,062 (8.25%) mobile sites, the Async Clipboard API (`writeText()` method) is one of the most used Fugu APIs. The `write()` method is used on 1,180 desktop and 1,227 mobile sites. The commercial website <a hreflang="en" href="https://clippingmagic.com/">Clipping Magic</a> allows you to remove the background of an image with the help of an AI algorithm. Just past ...
```
Same here, the Async Clipboard API is used in the same YouTube script (https://www.youtube.com/s/player/8d287e4d/player_ias.vflset/en_US/base.js).

.cc @tomayac

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,437 @@
+{{ figure_markup(
+image="fugu-podium.png",
+caption='The three websites that use the most Fugu APIs',
+description='The image shows a winner's podium with screenshots of the three websites that use the most Fugu APIs.',
```
```suggestion
description="The image shows a winner's podium with screenshots of the three websites that use the most Fugu APIs.'"
```

tunetheweb commented on src/content/en/2021/capabilities.md:
```diff
@@ -1,20 +1,437 @@
+The <a hreflang="en" href="https://www.chromium.org/teams/web-capabilities-fugu">Capabilities Project</a> (codename Fugu) is a joint effort by Microsoft, Intel, Google, and other Chromium contributors. It tries to bridge the gap between platform-specific applications and web apps by designing and implementing new powerful web platform APIs in a secure and privacy-preserving manner (see also the [ ...
+
+<figure>
+<blockquote>Project Fugu 🐡 is an effort to close gaps in the web's capabilities enabling new classes of applications to run on the web. […] APIs that Project Fugu is delivering enable new experiences on the web while preserving the web's core benefits of security, low-friction, and cross-platform delivery. All Project Fugu API proposals are made in the open and on the standards track.</blockquote ...
```
Never seen that `[...]` usage before?

```suggestion
<blockquote>Project Fugu 🐡 is an effort to close gaps in the web's capabilities enabling new classes of applications to run on the web… APIs that Project Fugu is delivering enable new experiences on the web while preserving the web's core benefits of security, low-friction, and cross-platform delivery. All Project Fugu API proposals are made in the open and on the standards track.</blockquote>
```

rviscomi replied:
The brackets are used to show that the original text was elided, as opposed to the original text literally containing the ellipsis.

tunetheweb replied:
Yeah I've always used a literal elipse for that under the assumption that always indicated parts of quotes.

Though I have used brackets when actually doing an edit to a quote:

> Rick said "The brackets are used to show that the original text was [edited], as opposed to the original text literally containing the ellipsis."

But anyway, happy to leave as is, if it's a known convention.

tunetheweb replied:
https://www.thepunctuationguide.com/ellipses.html#single-sentence

tunetheweb replied:
Seems like it's a controversial topic in editor circles:
https://english.stackexchange.com/questions/5972/how-are-bracket-ellipsis-used-in-quotations

The things you discover working on this project!

The Google Style Guide (that we use) doesn't seem to mandate square brackets: https://developers.google.com/style/ellipses?hl=en#ellipses-in-text

So I'd vote to remove them.

christianliebel commented:
@tunetheweb Alright, that should be it! 💪

tunetheweb commented:
Did a few more small edits: https://github.com/HTTPArchive/almanac.httparchive.org/pull/2398/commits/d739d46ed74fdc6dc47d87422fbe7093cfe683ba and https://github.com/HTTPArchive/almanac.httparchive.org/pull/2398/commits/cda681804b8a29c5882e315e691d4fc8a600f53e Hopefully nothing too controversial that you'll disagree with.

I've also updated the staging link: https://20211110t003629-dot-webalmanac.uk.r.appspot.com/en/2021/capabilities

One last question, do you think I should make the first use of an API in italics? From our Editor's Wiki:
> italics should be used for introducing terms.

I did it for the featured quotes to make them stand out more and now wondering about doing it more generally. We mostly do in the other chapters, though didn't in the 2020 Capabilities chapter. Anyone any strong opinions of this one way or another?

Other than that @rviscomi you happy to merge or wanna have a look?

rviscomi commented:
Ship it! Great work @christianliebel!

tunetheweb commented:
OK last commit: https://github.com/HTTPArchive/almanac.httparchive.org/pull/2398/commits/8b31abab47f7db79f23ff75e79d6c56f8a007e52

I went with the italics of the first use of the APIs, found one incorrect figure caption while doing that, and also added full stops to the captions.

Will merge once checks pass.

Thanks @christianliebel , @tomayac and @hemanth !

christianliebel commented:
@tunetheweb Excellent, all commits look great, thanks for adding the italics! It was a pleasure working with you again. 😊

tunetheweb commented:
And you @christianliebel ! Always one of the earliest chapters delivered which is much appreciated!

tomayac commented:
Thanks both, @christianliebel and @tunetheweb for the excellent work! Really an honor and pleasure working with you! `<3`
